server.js

```diff
@@ -21,9 +14,105 @@ app.use(express.json());
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.json({booksData});
+});
+
+// An alternative to get the array in pages 
+// e.g. http://localhost:8080/allbooks?page=1&limit=100
+app.get("/", paginatedResults(booksData), (req, res) => {
+  res.json(res.paginatedResults);
 });
 
+//filter books by typing author or title as a query parameter
+// e.g. http://localhost:8080/books?author=Bill%20Bryson
+app.get('/books', (req, res) => {
+	const { author, title } = req.query
+	let filteredBooks = booksData
+	
+	if(author) {
+    filteredBooks = filteredBooks.filter((item) => 
+    item.authors.toLocaleLowerCase()
+    .includes(author.toLocaleLowerCase()))
+  }
+
+	if(title) {
+    filteredBooks = filteredBooks.filter((item) => 
+    item.title.toLocaleLowerCase()
+    .includes(title.toLocaleLowerCase()))
+  }
+
+  if(filteredBooks.length === 0) {
+    return res.status(404).json({
+      success: false,
+      message: "Not Found",
+      body: {}
+    });
+  } else {
+      res.status(200).json({
+      success: true,
+      message: "OK",
+      body: {
+        booksData: filteredBooks
+      }
+    });
+  }
+
+
+  
+})
+
+// filter books by id
+// e.g. http://localhost:8080/books/23
+
+app.get("/books/:id", (req, res) => {
+  const singleBook = booksData.find((item) => item.bookID == req.params.id)
+  if (singleBook.length === 0) {
+    return res.status(404).json({
+      success: false,
+      message: "Not Found",
+      body: {}
+    });
+  } else {
+    res.status(200).json({
+      success: true,
+      message: "OK",
+      body: {
+        booksData: singleBook
+      }
+    })
+  }
+});
+
+function paginatedResults (model) {
+  return (req, res, next) => {
+    const page = +req.query.page 
+    const limit = +req.query.limit
```

Comment:
Great that you added this function, it's a nice feature to split up the data/info on more pages than one for the user (because that is what it's for right?) 
I'm not sure if it is possible to see here in backend?